client.py

```diff
@@ -41,22 +42,99 @@ def send_encrypted_msg(self, message: str):
+            self.receive_and_decrypt_msg()
+            self.sock.close()
+
     def receive_and_decrypt_msg(self):
         ciphertext = self.sock.recv(constants.CLIENT_PORT)
         plaintext = self.noise.decrypt(ciphertext)
         print(plaintext)
 
-    def run(self, message):
-        self.sock.connect(("localhost", constants.SERVER_PORT))
-        self.set_connection_keys()
-        self.noise_handshake()
-        self.send_encrypted_msg(message)
-        self.receive_and_decrypt_msg()
+    def register(self):
+        # TODO: read TPMs PCR values and somehow send them to a server
+        print("Registration complete, your identifier is <TOP_SECRET_PCR_HASH>")
+        return True
+
+    def run(self, ip, port, message):
+        self.ip = ip
+        self.port = port
+        if message:  # One time
+            self.sock.connect((ip, port))
+            self.set_connection_keys()
+            self.noise_handshake()
+            self.send_encrypted_msg(message)
+            self.receive_and_decrypt_msg()
+        else:  # Multiple messages
+            self.send_messages()
 
 
 if __name__ == "__main__":
-    client = Client()
-    if len(argv) > 1:
-        client.run(argv[1])
-    else:
-        print("Please write message as argument")
+    parser = argparse.ArgumentParser(
+        description="PV204 NoisyTPM - this is a part of team project for PV204. \
+                                                Client app can communicate with server using Noise framework \
+                                                and authenticate via TPM. Please see \
+                                                'https://github.com/Matej4545/PV204-NoisyTPM/' for more info."
+    )
+    parser.add_argument(
+        "-s",
+        "--server",
+        dest="server",
+        metavar="IP",
+        type=str,
+        nargs=1,
+        default="localhost",
+        help="An IP address or hostname of the server.",
+    )
+    parser.add_argument(
+        "-p",
+        "--port",
+        dest="port",
+        metavar="PORT",
+        type=int,
+        nargs=1,
+        default=5555,
+        help="A port where the server is listening.",
+    )
+    parser.add_argument(
+        "-m",
+        "--message",
+        metavar="MESSAGE",
+        dest="message",
+        type=str,
+        nargs="+",
+        help="Specify message as argument. For interactive session please omit.",
+    )
+    parser.add_argument(
+        "-r --register",
+        dest="register",
+        action="store_true",
+        default=False,
+        help="If you are not authenticated or running the app first time, you will need to register.",
+    )
+    args = parser.parse_args()
+
+    try:
+        message = "" if args.message is None else "".join(args.message).strip()
+        server = args.server[0].strip()
+        port = args.port[0]
+        client = Client()
+        if args.register:
+            client.register()
+        client.run(server, port, message)
```

Comment:
What about passing `server` and `port` to the init `Client` object? Is there some issue with that?